packages/block-library/src/navigation-link/style.scss

```diff
@@ -8,4 +8,10 @@
 		word-break: normal;
 		overflow-wrap: break-word;
 	}
+
+	// Hide the description by default.
+	// If a theme opts-in to show descriptions, they will need to provide styles for them.
+	.wp-block-navigation-item__description {
```

Comment:
The problem here is that themes were always opting-in to this feature, by creating their own nav-walkers... Displaying the description usually requires a lot of styling, which usually includes more than just font-sizes & colors. 
Descriptions are _usually_ an opinionated part of a theme's design, so after reading the discussion on https://github.com/WordPress/gutenberg/issues/28446#issuecomment-766307019 it made sense to just hide them and assume that if a theme wants to use description, they're going to add some styling to them - and that can also be used as a way to "opt-in" to the feature.

We _could_ of course add some default styles to descriptions and add a UI switch, but then opting-is is no longer left to the theme, but to users, and that would be a departure from previously established behavior. It's not a bad thing, just different and I don't know if that's something we want to do 🤔

Reply:
> We could of course add some default styles to descriptions and add a UI switch, but then opting-is is no longer left to the theme, but to users, and that would be a departure from previously established behavior. It's not a bad thing, just different and I don't know if that's something we want to do ...

I think in general, block themes are already staying from established behavior this way. And for the most part, that's a good thing! FSE is frequently handing more power to users, and preventing features from disappearing on theme switch. So allowing users more control here is definitely in line with those goals. 

To that point, I think the larger question here is: Do we want to support this feature in general? And if so, how would we re-design this to work well alongside other Nav block features? My general impression of this PR is that it seems generally harmless, so I'm not sure whether we need to answer that immediately.

Reply:
> To that point, I think the larger question here is: Do we want to support this feature in general?

I think we _have to_ support it... Without it, designs like the one in the screenshot below won't be possible
<img width="1161" alt="screenshot of the submenu of the gov.uk site using descriptions for menu items" src="https://user-images.githubusercontent.com/588688/160830846-fb6f717a-0496-4e8e-bf93-e782fd069ef4.png">
The screenshot above is one of many examples (this example is from the gov.uk site) where the menu descriptions are essential, both from a design perspective as well as from a UX perspective so that visitors can better comprehend the context of the menu items

Reply:
Yeah, let me rephrase the question: 

Do we want to support this feature *_**in the UI**_* now? And if so, how would we re-design this to work well alongside other Nav block features? 

The alternative of course is basically just what you're doing in the PR already: making this available, but leaving the more intense work for later.

Reply:
😅 apologies, I didn't understand what you meant initially.
Since the field is already there, I believe it would make sense to - as a first step - just make the existing field work.
I think it would be OK to tackle this in stages:
1. Make the existing description field work
2. Integrate the description better in the editor UI
3. Default styling for descriptions

This one is a relatively harmless PR just to get things started, the 1st step towards a "proper" implementation of the descriptions. It doesn't all have to be done in 1 go 😉

Reply:
That makes sense. I think the only risk is that themes would build in custom support for this feature, and then when we get to step 3, our default styles might conflict. But I suppose we can just be careful about specificity when we get to that step. 😅